state_tracker.go

```diff
@@ -160,65 +77,68 @@ func newSpeedLogRing(speedLogCount int) *ring.Ring {
 	return speedLog
 }
 
-type BinlogStateTracker struct {
-	*sync.RWMutex
-	lastProcessedBinlogPosition mysql.Position
-}
+type StateTracker struct {
+	BinlogRWMutex *sync.RWMutex
+	CopyRWMutex   *sync.RWMutex
 
-func NewBinlogStateTracker() *BinlogStateTracker {
-	return &BinlogStateTracker{
-		RWMutex:                     &sync.RWMutex{},
-		lastProcessedBinlogPosition: mysql.Position{},
-	}
-}
+	lastWrittenBinlogPosition                 mysql.Position
+	lastStoredBinlogPositionForInlineVerifier mysql.Position
```

Comment:
nit: `lastVerifiedBinlogPosition` is simpler? Do we need to introduce this in this PR? Seems to me like a followup to https://github.com/Shopify/ghostferry/pull/122

Reply:
It's not quite the last verified binlog position. As soon as the binlog event is added to the reverify store, we record this position. The reverify store will be dumped anyway so we won't lose anything. 

Either way, I'll remove it for now and send a follow up PR after #122.